CIP-meta-assets (ERC20-like assets)/README.md

```diff
@@ -0,0 +1,578 @@
+
+Additional checks regarding the state are left to the implementation.
+
+#### Account State Update
+
+```mermaid
+flowchart LR
+
+    subgraph transaction
+        .[ ]
+        style . fill:#FFFFFF00, stroke:#FFFFFF00;
+    end
+
+    stateManagerContract[state manager]
+    stateManager[state manager]
+
+    stateManager --o transaction 
+    --o stateManagerContract
+```
+
+The purpose of this transaction is to modify the state of an account.
+
+The only two rules regarding this transaction are:
+
+1) The NFT present in the input MUST be preserved in the output going back to the `stateManager`
+2) the credentials (first field of the datum) associated to an NFT MUST be preserved for that NFT (ie. the output having the nft has `Account` datum with first field equal to the first field of the input datum). 
+
+#### Minting Tokens
+
+```mermaid
+flowchart LR
+
+    subgraph transaction
+        .[ ]
+        style . fill:#FFFFFF00, stroke:#FFFFFF00;
+    end
+
+    stateManager[state manager]
+    transferManagerPolicy[(transfer manager policy)]
+    transferManagerContract[transfer manager]
+
+    stateManager -. receiver account state .-o transaction
+
+    transferManagerPolicy -. mints CNTs .->  transaction
+    -- minted CNTs --o transferManagerContract
+
+```
+
+The "Minting Tokens" transaction allows to create new programmable tokens.
+
+Programmable tokens are normal CNTs that are only present on `transferManager`
+contract with same hash as their own policy.
+
+If, by incorrect implementation of the standard, these tokens are found outside the respective transfer manager, they are no longer to be considered as programmable tokens.
+
+The transaction MUST have an output going to the `transferManager` contract with the correct [`TransferManagerDatum`](#TransferManagerDatum) **inline** datum.
+ 
+The first field of the datum MUST be the name of the NFT present on the reference input coming from the state manager.
+
+Additional, implementation-specific, arbitrary logic (eg. capped supply, or one-time mints, etc. ) MAY be added.
+
+#### Burning Tokens
+
+```mermaid
+flowchart LR
+
+    subgraph transaction
+        .[ ]
+        style . fill:#FFFFFF00, stroke:#FFFFFF00;
+    end
+
+    stateManager[state manager]
+    transferManagerContract[transfer manager]
+
+    stateManager -. sender account state .-o transaction
+
+    transferManagerContract -- Burn Redeemer --o  transaction -..-x a[ ]
+
+    style a fill:#FFFFFF00, stroke:#FFFFFF00;
+```
+
+If an utxo in the `transferManager` is spent with `Burn` redeemer, the **entire** amount of the programmable token on that utxo MUST be burnt.
+
+Only **one (1)** utxo from the `transferManager` is allowed to be spent in a burn transaction.
+
+If a user desires to burn a different amount than the one currently present they should break the operation in 2 different transactions:
```

Comment:
Why only one utxo per transaction? This seems unnecessarily restrictive. Why can't multiple utxos be spent? The entire balance from each utxo spent can be burned. The observer can tell which utxos are being used for burning based on the redeemers used to spend them.